nixos/modules/services/torrent/flood.nix

```diff
@@ -0,0 +1,115 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.flood;
+in
+{
+  meta.maintainers = with lib.maintainers; [ thiagokokada ];
+
+  options.services.flood = {
+    enable = lib.mkEnableOption "flood";
+    package = lib.mkPackageOption pkgs "flood" { };
+    openFirewall = lib.mkEnableOption "" // {
+      description = "Whether to automatically open the necessary ports in the firewall.";
+      default = true;
+    };
+    user = lib.mkOption {
+      type = lib.types.str;
+      description = "User account used to run flood.";
+      default = "flood";
+    };
+    group = lib.mkOption {
+      type = lib.types.str;
+      description = "Group account used to run flood.";
+      default = "flood";
+    };
+    port = lib.mkOption {
+      type = lib.types.int;
+      description = "Port to bind webserver.";
+      default = 3000;
+    };
+    host = lib.mkOption {
+      type = lib.types.str;
+      description = "Host to bind webserver.";
+      default = "0.0.0.0";
```

Comment:
- Can this bind dualtstacked? For example using `::`
- Can we default to localhost, just so people don't accidentally expose this?

Reply:
- Not sure.
- Yes, good idea.